### ISSUE TITLE
Integrate KMM to enable support for iOS

### ISSUE THREAD

**Issue by peakvalleytech** [alisonthemonster/Presently#209]
I know there have been discussions about having an iOS version. I have been looking at KMM and if used can help us create an iOS version without the need to create a completely separate iOS version. I won't go into detail about KMM, but, in short, it will allow writing many parts in almost every part actually, except for very platform-specific parts like the UI. 

Also, I realize that this app has already been ripped off and put on the AppStore. My question is why do you remove it? Was it because it failed to honor the OSS license. Because I realize it was 'ripped off' but in the OSS community, this is acceptable if not completey encouraged. 

I would like to get started on an iOS version. Is that okay with you? @alisonthemonster

**Comment by alisonthemonster** [comment]
Hi there! 

As you've probably noticed I'm not the most responsive lately. I really don't think I have the capacity to support an entirely new codebase at the moment. I've tried to keep Presently for Android at the highest standards I possibly can, and doing so in an entirely new repo, with a language/framework I don't know well just doesn't seem feasible at the moment. From setting up CI, code coverage, crash reporting, following best practices and using modern architectures/patterns, etc I just know I can't commit fully to working on this now. 

I also want there to be new designs created specifically for iOS, which I haven't had the time or resources to do. 

As for the copy that was put in the store, I had no control over it. When/if a Presently iOS client is ready to release, I would want to own the store listing, the developer console, the crash reports, the analytics, the release cycle, etc. The knockoff was also a bit glitchy and had some issues. It was also closed source so we don't know what they had done to make it. 
As for it being acceptable for someone to create a copy in the OSS world, this is not the case. This app has no open source license, which means default copyright rules apply, from [Github's guidelines](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/licensing-a-repository#choosing-the-right-license):
> You're under no obligation to choose a license. However, without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work.

So in summary I just don't think I have the capacity to properly support an iOS client at the time. I love your enthusiasm, but I just don't have the time to create the Presently iOS client that I want to build.

**Comment by peakvalleytech** [comment]
Hi there, 
I appreciate your response. And any response is good. I completely understand. I have my own projects and I can't always work on them when I want to. 

Thanks for clarifying that issue regarding the 'iOS knockoff'.  

I'm actually working on an OSS iOS version. I have most of the basic UI functionality done. 

I was wondering if you would be okay in collaborating on the iOS version. I'm fine with you owning the repo. I would just like to be able to send my PR. Maybe you can set up a repository and I can send a PR. That way we will have a base for the codebase. And you can start setting up the CI, crash reporting, console, architechture, etc.

Or I could just abandon the project and maybe release myself. But, of course, not a derivative. It would have a different name and of course, the code would be different, the design too, and I will try not to copy very specific details unique to that Prsently. It would just be another gratitude journal app, but for iOS.
